reframe/core/buildsystems.py

```diff
@@ -661,42 +661,52 @@ class EasyBuild(BuildSystem):
     options = fields.TypedField(typ.List[str])
     emit_package = fields.TypedField(bool)
     package_opts = fields.TypedField(typ.Dict[str, str])
+    srcdir = fields.TypedField(str, type(None))
 
     def __init__(self):
         super().__init__()
         self.easyconfigs = []
         self.options = []
         self.emit_package = False
-        self.eb_modules = []
+        self.package_opts = {}
+        self._eb_modules = []
+        self.srcdir = None
 
     def emit_build_commands(self, environ):
         easyconfigs = ' '.join(self.easyconfigs)
-
         if self.emit_package:
             self.options.append('--package')
             for key, val in self.package_opts.items():
-                self.options.append(f"--package-{key}={val}")
+                self.options.append(f'--package-{key}={val}')
 
         options = ' '.join(self.options)
 
-        cmd = [f"EASYBUILD_BUILDPATH={self._eb_sandbox}/build \\",
-               f"EASYBUILD_INSTALLPATH={self._eb_sandbox} \\",
-               f"EASYBUILD_PREFIX={self._eb_sandbox} \\",
-               f"EASYBUILD_SOURCEPATH={self._eb_sandbox} \\",
-               f"eb {easyconfigs} {options}"]
+        self._eb_sandbox = os.path.join(self.srcdir, 'rfm_easybuild')
 
-        return cmd
+        return [f'EASYBUILD_BUILDPATH={self._eb_sandbox}/build \\',
+                f'EASYBUILD_INSTALLPATH={self._eb_sandbox} \\',
+                f'EASYBUILD_PREFIX={self._eb_sandbox} \\',
+                f'EASYBUILD_SOURCEPATH={self._eb_sandbox} \\',
+                f'eb {easyconfigs} {options}']
```

Comment:
Why don't you use `export EASYBUILD_XXX` here?

Reply:
Ignore this comment. It has to be in a single line.

Reply:
Actually, you can generate multiple lines, so you should be able to use `export`s here.